signer/src/transaction_coordinator.rs

```diff
@@ -950,13 +953,28 @@ where
         let bitcoin_client = self.context.get_bitcoin_client();
         let fee_rate = bitcoin_client.estimate_fee_rate().await?;
 
+        // Retrieve the signer's current UTXO.
         let utxo = self
             .context
             .get_storage()
             .get_signer_utxo(chain_tip, aggregate_key, self.context_window)
             .await?
             .ok_or(Error::MissingSignerUtxo)?;
-        let last_fees = bitcoin_client.get_last_fee(utxo.outpoint).await?;
+
+        // Retrieve the last sweep package for the above UTXO. These are transactions
+        // which exist in the mempool.
+        let last_sweep_package = self
+            .context
+            .get_storage()
+            .get_sweep_transaction_package(&utxo.outpoint.txid.into())
+            .await?;
+
+        // If the last sweep package is empty, then we don't have any fees to
+        // report, so we return None. Otherwise, we return the calculated fees
+        // from the last sweep package.
+        let last_fees = last_sweep_package
+            .is_empty()
+            .then(|| last_sweep_package.get_fees());
```

Comment:
I think this is backwards. If `last_sweep_package.is_empty()` then `last_fees` should be `None`

Reply:
I wasn't done! :stuck_out_tongue_closed_eyes: But yeah you're right, refactoring gone bad. But now there's tests that catch this and verify the whole `get_btc_state()` function both when sweeps exist and when they don't.
[6cf7d78](https://github.com/stacks-network/sbtc/pull/810/commits/6cf7d789d6db074533b7484aca5902d9eb310a99)

Reply:
Oh whoops, I didn't check to see if you were done yet.